cmp/grid/GridModel.js

```diff
@@ -908,7 +908,12 @@ export class GridModel extends HoistModel {
         const omit = isFunction(config.omit) ? config.omit() : config.omit;
         if (omit) return null;
 
-        if (config.children) return new ColumnGroup(config, this);
+        if (config.children) {
+            config.children = compact(config.children.map(c => this.buildColumn(c)));
```

Comment:
this mutates the config -- what if it was going to be used again for another grid?